src/v/coproc/script_context.cc

```diff
@@ -65,29 +69,49 @@ script_context::script_context(
       "valid subscription list");
 }
 
+static void exc_printer(const std::exception_ptr& eptr) {
+    try {
+        /// If an exception is unhandled by this method, it will percolate to
+        /// the caller of start(). ATM by design only one exception should do
+        /// this, the \ref script_failed_exception.
+        std::rethrow_exception(eptr);
+    } catch (const std::logic_error& e) {
+        vlog(
+          coproclog.error,
+          "std::logic_error observed wasm engine sent unexpected reply, "
+          "default behavior is to retry, exception: {}",
+          e.what());
+    } catch (const std::runtime_error& e) {
+        vlog(
+          coproclog.warn,
+          "Error observed with RPC to/from wasm engine, behavior is to retry, "
+          "exception: {}",
+          e.what());
+    } catch (const ss::sleep_aborted& e) {
+        vlog(
+          coproclog.debug,
+          "Sleep aborted early due to context shutdown() {}",
+          e.what());
+    }
+}
+
 ss::future<> script_context::start() {
     vassert(
       !_gate.is_closed() && _gate.get_count() == 0,
       "Cannot call start() twice");
-    (void)ss::with_gate(_gate, [this] {
+    return ss::with_gate(_gate, [this] {
```

Comment:
Can you add some reasoning to the commit message for this change.

Reply:
Yeah this is me ad-libbing. Removing

Reply:
Ok now I remember why I did this. I will explain in the commit msg. I needed the caller of start to know when the future resolved, re-adding